vdb/flu_upload.py

```diff
@@ -94,6 +94,9 @@ def parse_fasta_file(self, fasta, **kwargs):
         else:
             for record in SeqIO.parse(handle, "fasta"):
                 content = list(map(lambda x: x.strip(), record.description.replace(">", "").split('|')))
+                # Only check the first record since we expect a single file to have the same headers
+                if len(sequences) == 0 and len(content) < len(sequence_fasta_fields):
+                    raise Exception(f"FASTA header only has {len(content)} fields but we are expecting {len(sequence_fasta_fields)} fields.")
```

Comment:
_from https://github.com/nextstrain/fauna/pull/182#discussion_r2601047980_:
> If the sequence header metadata lacks these fields, we'll get a key error in transfer_fields which should be easy to track down.

I realized this was not the case when I tested with downloads that lacked the expected fields. It would just silently set those fields as empty strings. I've added a check for both the XLS and FASTA files that they should have all expected fields. This would flag issues earlier rather than later and force us to change the fauna code to match the new fields.